lib/Translation/ExportVerilog/ExportVerilogInternals.h

```diff
@@ -41,10 +41,20 @@ struct ModuleNameManager {
   }
 
   StringRef getName(Value value) { return getName(ValueOrOp(value)); }
-  StringRef getName(Operation *op) { return getName(ValueOrOp(op)); }
+  StringRef getName(Operation *op) {
+    // If this operation has a result, get the name of the result.
+    if (op->getNumResults() == 1 && hasName(op->getResult(0)))
+      return getName(op->getResult(0));
```

Comment:
Please don't implicitly look up all stuff with a name attribute.  We need to get away from this and handle explicit / specific things like reg/wire.

Reply:
Special cased it for `RegOp` and `WireOp`. I think these are the only two operations we need to consider.

Reply:
This is a weird artifact of attribute op arguments and general attributes on an op sharing a namespace.  I was pondering a "Namable" operation trait to simplify these kinds of checks.